docs/guides/nextauth-migration.mdx

```diff
@@ -0,0 +1,362 @@
+
+```tsx filename="app/layout.tsx"
+import { ClerkProvider } from '@clerk/nextjs'
+import './globals.css'
+ 
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  return (
+    <ClerkProvider>
+      <html lang="en">
+        <body>{children}</body>
+      </html>
+    </ClerkProvider>
+  )
+}
+```
+
+```tsx filename="_app.tsx"
+import '@/styles/globals.css'
+import { ClerkProvider } from "@clerk/nextjs";
+import type { AppProps } from "next/app";
+function MyApp({ Component, pageProps }: AppProps) {
+  return (
+    <ClerkProvider {...pageProps}>
+      <Component {...pageProps} />
+    </ClerkProvider>
+  );
+}
+export default MyApp;
+```
+</CodeBlockTabs>
+
+### Setup Sign-up and sign-in UI
+
+#### Account Portal
+
+Account Portal is the fastest way to authenticate your app. Clerk's Account Portal hosts the `<SignIn />`, `<SignUp />`, `<UserProfile />` and other components for your application. Read more about [Account Portal](docs/account-portal/getting-started). 
+
+To use the Account Portal, remove the routes that mount the NextAuth.js sign-in and sign-up UI. Replace the links to those routes with the [`<SignInButton>`](/docs/components/unstyled/sign-in-button) or [`<SignUpButton>`](/docs/components/unstyled/sign-out-button) components.
+
+#### Self-hosted UI
+
+If you want sign-in and sign-up pages in your own application, you can use [Clerk's pre-built components](/docs/references/nextjs/custom-signup-signin-pages), or implement the UI yourself using [Clerk's hooks](/docs/custom-flows/overview).
+
+#### Custom UI
+
+For fully custom UI, replace the existing functionality with [appropriate custom flows](/docs/custom-flows/overview#custom-flows). With Custom flows, you leverage Clerk's React hooks `useSignIn` and `useSignUp` to build fully custom UI that handles authentication with email and password, OAuth, magic links, OTP or any other authentication option Clerk offers.
+
+### Protect your app
+
+#### Middleware
+
+You will need to remove the NextAuth.js Middleware from your application, and replace it with the Clerk Middleware helper. NextAuth's middleware always rejects unauthorized requests. You may have additionally configured the Next.js middleware config to protect specific private routes. You will need to make note of this configuration so you can recreate it.
+
+##### Remove the NextAuth.js Middleware
+
+```tsx filename="middleware.ts"
+export { default } from "next-auth/middleware"
+
+export const config = { matcher: [/** private routes */] }
+```
+##### Add Clerk's `authMiddleware()`
+
+The example below is the starting Middleware configuration. Please read the [`authMiddleware()`](docs/references/nextjs/auth-middleware) documentation to learn how to configuration your Middleware. Clerk's Middleware will, by default, run for your entire application and gives you fine-grained control over handling the authenticated state. 
```

Comment:
```suggestion
The example below is the starting Middleware configuration. Please read the [`authMiddleware()`](/docs/references/nextjs/auth-middleware) documentation to learn how to configuration your Middleware. Clerk's Middleware will, by default, run for your entire application and gives you fine-grained control over handling the authenticated state. 
```

Reply:
again, did the link work or was that breaking it?